From 281274d71800333bb067fb7700dfe0c74d911431 Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Wed, 14 Aug 2019 18:31:38 +0800
Subject: [PATCH] Rule sysctl_net_ipv4_conf_default_send_redirects add oval
 check

For rule sysctl_net_ipv4_conf_default_send_redirects add oval check

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../oval/wrlinux1019.xml                           | 15 ++++++
 .../oval/wrlinux1019.xml                           | 27 ++++++++++
 .../rule.yml                                       | 44 ++++++++++++++++
 .../oval/wrlinux1019.xml                           | 61 ++++++++++++++++++++++
 .../rule.yml                                       | 44 ++++++++++++++++
 5 files changed, 191 insertions(+)
 create mode 100644 linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_net_ipv4_conf_default_send_redirects/oval/wrlinux1019.xml
 create mode 100644 linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_runtime_net_ipv4_conf_default_send_redirects/oval/wrlinux1019.xml
 create mode 100644 linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_runtime_net_ipv4_conf_default_send_redirects/rule.yml
 create mode 100644 linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_static_net_ipv4_conf_default_send_redirects/oval/wrlinux1019.xml
 create mode 100644 linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_static_net_ipv4_conf_default_send_redirects/rule.yml

diff --git a/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_net_ipv4_conf_default_send_redirects/oval/wrlinux1019.xml b/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_net_ipv4_conf_default_send_redirects/oval/wrlinux1019.xml
new file mode 100644
index 0000000..04e7df5
--- /dev/null
+++ b/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_net_ipv4_conf_default_send_redirects/oval/wrlinux1019.xml
@@ -0,0 +1,15 @@
+<def-group>
+  <definition class="compliance" id="sysctl_net_ipv4_conf_default_send_redirects" version="3">
+    <metadata>
+      <title>Kernel "net.ipv4.conf.default.send_redirects" Parameter Configuration and Runtime Check</title>
+      <affected family="unix">
+        <platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>The "net.ipv4.conf.default.send_redirects" kernel parameter should be set to the appropriate value in both system configuration and system runtime.</description>
+    </metadata>
+    <criteria operator="AND">
+      <extend_definition comment="net.ipv4.conf.default.send_redirects configuration setting check" definition_ref="sysctl_static_net_ipv4_conf_default_send_redirects" />
+      <extend_definition comment="net.ipv4.conf.default.send_redirects runtime setting check" definition_ref="sysctl_runtime_net_ipv4_conf_default_send_redirects" />
+    </criteria>
+  </definition>
+</def-group>
diff --git a/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_runtime_net_ipv4_conf_default_send_redirects/oval/wrlinux1019.xml b/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_runtime_net_ipv4_conf_default_send_redirects/oval/wrlinux1019.xml
new file mode 100644
index 0000000..8424fa8
--- /dev/null
+++ b/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_runtime_net_ipv4_conf_default_send_redirects/oval/wrlinux1019.xml
@@ -0,0 +1,27 @@
+<def-group>
+  <definition class="compliance" id="sysctl_runtime_net_ipv4_conf_default_send_redirects" version="3">
+    <metadata>
+      <title>Kernel "net.ipv4.conf.default.send_redirects" Parameter Runtime Check</title>
+      <affected family="unix">
+        <platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>The kernel "net.ipv4.conf.default.send_redirects" parameter should be set to "0" in system runtime.</description>
+    </metadata>
+    <criteria operator="AND">
+      <criterion comment="kernel runtime parameter net.ipv4.conf.default.send_redirects set to 0" test_ref="test_runtime_sysctl_net_ipv4_conf_default_send_redirects" />
+    </criteria>
+  </definition>
+
+  <unix:sysctl_test check="all" check_existence="all_exist" comment="kernel runtime parameter net.ipv4.conf.default.send_redirects set to 0" id="test_runtime_sysctl_net_ipv4_conf_default_send_redirects" version="1">
+    <unix:object object_ref="object_sysctl_net_ipv4_conf_default_send_redirects" />
+    <unix:state state_ref="state_sysctl_net_ipv4_conf_default_send_redirects" />
+  </unix:sysctl_test>
+
+  <unix:sysctl_object id="object_sysctl_net_ipv4_conf_default_send_redirects" version="1">
+    <unix:name>net.ipv4.conf.default.send_redirects</unix:name>
+  </unix:sysctl_object>
+
+  <unix:sysctl_state id="state_sysctl_net_ipv4_conf_default_send_redirects" version="1">
+    <unix:value datatype="int" operation="equals">0</unix:value>
+  </unix:sysctl_state>
+</def-group>
diff --git a/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_runtime_net_ipv4_conf_default_send_redirects/rule.yml b/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_runtime_net_ipv4_conf_default_send_redirects/rule.yml
new file mode 100644
index 0000000..33722a8
--- /dev/null
+++ b/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_runtime_net_ipv4_conf_default_send_redirects/rule.yml
@@ -0,0 +1,44 @@
+documentation_complete: true
+
+prodtype: wrlinux1019
+
+title: 'Kernel "net.ipv4.conf.default.send_redirects" Parameter Runtime Check'
+
+description: '{{{ describe_sysctl_option_value(sysctl="net.ipv4.conf.default.send_redirects", value="0") }}}'
+
+rationale: |-
+    ICMP redirect messages are used by routers to inform hosts that a more
+    direct route exists for a particular destination. These messages contain information
+    from the system's route table possibly revealing portions of the network topology.
+    <br />
+    The ability to send ICMP redirects is only appropriate for systems acting as routers.
+
+severity: medium
+
+identifiers:
+    cce@rhel6: 27001-7
+    cce@rhel7: 80999-6
+    cce@rhel8: 80921-0
+
+references:
+    anssi: NT28(R22)
+    stigid@rhel6: RHEL-06-000080
+    srg@rhel6: SRG-OS-999999
+    disa@rhel6: '1551'
+    cis: 3.1.2
+    cjis: 5.10.1.1
+    cui: 3.1.20
+    disa: "366"
+    nist: AC-4,CM-7,SC-5,SC-7
+    nist-csf: DE.AE-1,DE.CM-1,ID.AM-3,PR.AC-5,PR.DS-4,PR.DS-5,PR.IP-1,PR.PT-3,PR.PT-4
+    srg: SRG-OS-000480-GPOS-00227
+    stigid@rhel7: "040650"
+    isa-62443-2013: 'SR 1.1,SR 1.10,SR 1.11,SR 1.12,SR 1.13,SR 1.2,SR 1.3,SR 1.4,SR 1.5,SR 1.6,SR 1.7,SR 1.8,SR 1.9,SR 2.1,SR 2.2,SR 2.3,SR 2.4,SR 2.5,SR 2.6,SR 2.7,SR 3.1,SR 3.5,SR 3.8,SR 4.1,SR 4.3,SR 5.1,SR 5.2,SR 5.3,SR 6.2,SR 7.1,SR 7.2,SR 7.6'
+    isa-62443-2009: 4.2.3.4,4.3.3.4,4.3.3.5.1,4.3.3.5.2,4.3.3.5.3,4.3.3.5.4,4.3.3.5.5,4.3.3.5.6,4.3.3.5.7,4.3.3.5.8,4.3.3.6.1,4.3.3.6.2,4.3.3.6.3,4.3.3.6.4,4.3.3.6.5,4.3.3.6.6,4.3.3.6.7,4.3.3.6.8,4.3.3.6.9,4.3.3.7.1,4.3.3.7.2,4.3.3.7.3,4.3.3.7.4,4.3.4.3.2,4.3.4.3.3,4.4.3.3
+    cobit5: APO01.06,APO13.01,BAI04.04,BAI10.01,BAI10.02,BAI10.03,BAI10.05,DSS01.03,DSS01.05,DSS03.01,DSS03.05,DSS05.02,DSS05.04,DSS05.05,DSS05.07,DSS06.02,DSS06.06
+    iso27001-2013: A.10.1.1,A.11.1.4,A.11.1.5,A.11.2.1,A.12.1.1,A.12.1.2,A.12.1.3,A.12.5.1,A.12.6.2,A.13.1.1,A.13.1.2,A.13.1.3,A.13.2.1,A.13.2.2,A.13.2.3,A.13.2.4,A.14.1.2,A.14.1.3,A.14.2.2,A.14.2.3,A.14.2.4,A.17.2.1,A.6.1.2,A.7.1.1,A.7.1.2,A.7.3.1,A.8.2.2,A.8.2.3,A.9.1.1,A.9.1.2,A.9.2.3,A.9.4.1,A.9.4.4,A.9.4.5
+    cis-csc: 1,11,12,13,14,15,16,18,2,3,4,6,7,8,9
+
+{{{ complete_ocil_entry_sysctl_option_value(sysctl="net.ipv4.conf.default.send_redirects", value="0") }}}
+
+platform: machine
diff --git a/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_static_net_ipv4_conf_default_send_redirects/oval/wrlinux1019.xml b/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_static_net_ipv4_conf_default_send_redirects/oval/wrlinux1019.xml
new file mode 100644
index 0000000..ab6117c
--- /dev/null
+++ b/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_static_net_ipv4_conf_default_send_redirects/oval/wrlinux1019.xml
@@ -0,0 +1,61 @@
+<def-group>
+  <definition class="compliance" id="sysctl_static_net_ipv4_conf_default_send_redirects" version="3">
+    <metadata>
+      <title>Kernel "net.ipv4.conf.default.send_redirects" Parameter Configuration Check</title>
+      <affected family="unix">
+        <platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>The kernel "net.ipv4.conf.default.send_redirects" parameter should be set to "0" in the system configuration.</description>
+    </metadata>
+    <criteria operator="OR">
+      <criterion comment="kernel static parameter net.ipv4.conf.default.send_redirects set to 0 in /etc/sysctl.conf" test_ref="test_static_sysctl_net_ipv4_conf_default_send_redirects" />
+      <!-- see sysctl.d(5) -->
+      <criterion comment="kernel static parameter net.ipv4.conf.default.send_redirects set to 0 in /etc/sysctl.d/*.conf" test_ref="test_static_etc_sysctld_net_ipv4_conf_default_send_redirects" />
+      <criterion comment="kernel static parameter net.ipv4.conf.default.send_redirects set to 0 in /run/sysctl.d/*.conf" test_ref="test_static_run_sysctld_net_ipv4_conf_default_send_redirects" />
+      <criterion comment="kernel static parameter net.ipv4.conf.default.send_redirects set to 0 in /usr/lib/sysctl.d/*.conf" test_ref="test_static_usr_lib_sysctld_net_ipv4_conf_default_send_redirects" />
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="net.ipv4.conf.default.send_redirects static configuration" id="test_static_sysctl_net_ipv4_conf_default_send_redirects" version="1">
+    <ind:object object_ref="object_static_sysctl_net_ipv4_conf_default_send_redirects" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_test check="all" comment="net.ipv4.conf.default.send_redirects static configuration in /etc/sysctl.d/*.conf" id="test_static_etc_sysctld_net_ipv4_conf_default_send_redirects" version="1">
+    <ind:object object_ref="object_static_etc_sysctld_net_ipv4_conf_default_send_redirects" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_test check="all" comment="net.ipv4.conf.default.send_redirects static configuration in /etc/sysctl.d/*.conf" id="test_static_run_sysctld_net_ipv4_conf_default_send_redirects" version="1">
+    <ind:object object_ref="object_static_run_sysctld_net_ipv4_conf_default_send_redirects" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_test check="all" comment="net.ipv4.conf.default.send_redirects static configuration in /etc/sysctl.d/*.conf" id="test_static_usr_lib_sysctld_net_ipv4_conf_default_send_redirects" version="1">
+    <ind:object object_ref="object_static_usr_lib_sysctld_net_ipv4_conf_default_send_redirects" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="object_static_sysctl_net_ipv4_conf_default_send_redirects" version="1">
+    <ind:filepath>/etc/sysctl.conf</ind:filepath>
+    <ind:pattern operation="pattern match">^[\s]*net.ipv4.conf.default.send_redirects[\s]*=[\s]*0[\s]*$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_object id="object_static_etc_sysctld_net_ipv4_conf_default_send_redirects" version="1">
+    <ind:path>/etc/sysctl.d</ind:path>
+    <ind:filename operation="pattern match">^.*\.conf$</ind:filename>
+    <ind:pattern operation="pattern match">^[\s]*net.ipv4.conf.default.send_redirects[\s]*=[\s]*0[\s]*$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_object id="object_static_run_sysctld_net_ipv4_conf_default_send_redirects" version="1">
+    <ind:path>/run/sysctl.d</ind:path>
+    <ind:filename operation="pattern match">^.*\.conf$</ind:filename>
+    <ind:pattern operation="pattern match">^[\s]*net.ipv4.conf.default.send_redirects[\s]*=[\s]*0[\s]*$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_object id="object_static_usr_lib_sysctld_net_ipv4_conf_default_send_redirects" version="1">
+    <ind:path>/usr/lib/sysctl.d</ind:path>
+    <ind:filename operation="pattern match">^.*\.conf$</ind:filename>
+    <ind:pattern operation="pattern match">^[\s]*net.ipv4.conf.default.send_redirects[\s]*=[\s]*0[\s]*$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+</def-group>
diff --git a/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_static_net_ipv4_conf_default_send_redirects/rule.yml b/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_static_net_ipv4_conf_default_send_redirects/rule.yml
new file mode 100644
index 0000000..5a9cca8
--- /dev/null
+++ b/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_static_net_ipv4_conf_default_send_redirects/rule.yml
@@ -0,0 +1,44 @@
+documentation_complete: true
+
+prodtype: wrlinux1019
+
+title: 'Kernel "net.ipv4.conf.default.send_redirects" Parameter Configuration Check'
+
+description: '{{{ describe_sysctl_option_value(sysctl="net.ipv4.conf.default.send_redirects", value="0") }}}'
+
+rationale: |-
+    ICMP redirect messages are used by routers to inform hosts that a more
+    direct route exists for a particular destination. These messages contain information
+    from the system's route table possibly revealing portions of the network topology.
+    <br />
+    The ability to send ICMP redirects is only appropriate for systems acting as routers.
+
+severity: medium
+
+identifiers:
+    cce@rhel6: 27001-7
+    cce@rhel7: 80999-6
+    cce@rhel8: 80921-0
+
+references:
+    anssi: NT28(R22)
+    stigid@rhel6: RHEL-06-000080
+    srg@rhel6: SRG-OS-999999
+    disa@rhel6: '1551'
+    cis: 3.1.2
+    cjis: 5.10.1.1
+    cui: 3.1.20
+    disa: "366"
+    nist: AC-4,CM-7,SC-5,SC-7
+    nist-csf: DE.AE-1,DE.CM-1,ID.AM-3,PR.AC-5,PR.DS-4,PR.DS-5,PR.IP-1,PR.PT-3,PR.PT-4
+    srg: SRG-OS-000480-GPOS-00227
+    stigid@rhel7: "040650"
+    isa-62443-2013: 'SR 1.1,SR 1.10,SR 1.11,SR 1.12,SR 1.13,SR 1.2,SR 1.3,SR 1.4,SR 1.5,SR 1.6,SR 1.7,SR 1.8,SR 1.9,SR 2.1,SR 2.2,SR 2.3,SR 2.4,SR 2.5,SR 2.6,SR 2.7,SR 3.1,SR 3.5,SR 3.8,SR 4.1,SR 4.3,SR 5.1,SR 5.2,SR 5.3,SR 6.2,SR 7.1,SR 7.2,SR 7.6'
+    isa-62443-2009: 4.2.3.4,4.3.3.4,4.3.3.5.1,4.3.3.5.2,4.3.3.5.3,4.3.3.5.4,4.3.3.5.5,4.3.3.5.6,4.3.3.5.7,4.3.3.5.8,4.3.3.6.1,4.3.3.6.2,4.3.3.6.3,4.3.3.6.4,4.3.3.6.5,4.3.3.6.6,4.3.3.6.7,4.3.3.6.8,4.3.3.6.9,4.3.3.7.1,4.3.3.7.2,4.3.3.7.3,4.3.3.7.4,4.3.4.3.2,4.3.4.3.3,4.4.3.3
+    cobit5: APO01.06,APO13.01,BAI04.04,BAI10.01,BAI10.02,BAI10.03,BAI10.05,DSS01.03,DSS01.05,DSS03.01,DSS03.05,DSS05.02,DSS05.04,DSS05.05,DSS05.07,DSS06.02,DSS06.06
+    iso27001-2013: A.10.1.1,A.11.1.4,A.11.1.5,A.11.2.1,A.12.1.1,A.12.1.2,A.12.1.3,A.12.5.1,A.12.6.2,A.13.1.1,A.13.1.2,A.13.1.3,A.13.2.1,A.13.2.2,A.13.2.3,A.13.2.4,A.14.1.2,A.14.1.3,A.14.2.2,A.14.2.3,A.14.2.4,A.17.2.1,A.6.1.2,A.7.1.1,A.7.1.2,A.7.3.1,A.8.2.2,A.8.2.3,A.9.1.1,A.9.1.2,A.9.2.3,A.9.4.1,A.9.4.4,A.9.4.5
+    cis-csc: 1,11,12,13,14,15,16,18,2,3,4,6,7,8,9
+
+{{{ complete_ocil_entry_sysctl_option_value(sysctl="net.ipv4.conf.default.send_redirects", value="0") }}}
+
+platform: machine
-- 
1.9.1

